### PR TITLE
[Window Placement article] Add defensive permission code

### PR DESCRIPTION
**Pull request by tomayac**
Changes proposed in this pull request:

- Add defensive permission code.

### CODE REVIEW

File: site/en/articles/multi-screen-window-placement/index.md [hunk 1]
```diff
@@ -143,6 +144,50 @@ try {
...
+          state = 'Window management not supported.';
+        } else {
+          state = `${err.name}: ${err.message}`;
+        }
+      }
+    } else {
+      state = `${err.name}: ${err.message}`;
+    }
+  }
+  if ('request' in Permissions.prototype) {
```
Comment (beaufortfrancois) [inline site/en/articles/multi-screen-window-placement/index.md:182]: This API has been behind the experimental web platform features flag for years. Are there signs this is going to ship soon?

File: site/en/articles/multi-screen-window-placement/index.md [hunk 2]
```diff
@@ -143,6 +144,50 @@ try {
 }
 ```
 
+While browsers with the old and the new permission name are in use, be sure to use defensive code when requesting permission, as in the example below.
+
+```js
```
Comment (beaufortfrancois) [inline site/en/articles/multi-screen-window-placement/index.md:149]: What do you think of this?

```js
async function requestWindowManagementPermission() {
  let state;
  let name;
  // The new permission name.
  try {
    ({ state, name } = await navigator.permissions.query({
      name: "window-management",
    }));
  } catch (err) {
    if (err.name !== "TypeError") {
      return `${err.name}: ${err.message}`;
    }
    // The old permission name.
    try {
      ({ state, name } = await navigator.permissions.query({
        name: "window-placement",
      }));
    } catch (err) {
      if (err.name === "TypeError") {
        return "Window management not supported.";
      }
      return `${err.name}: ${err.message}`;
    }
  }
  if ("request" in Permissions.prototype) {
    ({ state } = await navigator.permissions.request({ name }));
  }
  return state;
}

document.querySelector("button").addEventListener("click", async () => {
  const state = await requestWindowManagementPermission();
  document.querySelector("pre").textContent = state;
});
```

File: site/en/articles/multi-screen-window-placement/index.md (hunk 1 above)
Comment (beaufortfrancois) [reply site/en/articles/multi-screen-window-placement/index.md:182]: If not, my previous proposal could be reduced even further:

```js
async function getWindowManagementPermissionState() {
  let state;
  // The new permission name.
  try {
    ({ state } = await navigator.permissions.query({
      name: "window-management",
    }));
  } catch (err) {
    if (err.name !== "TypeError") {
      return `${err.name}: ${err.message}`;
    }
    // The old permission name.
    try {
      ({ state } = await navigator.permissions.query({
        name: "window-placement",
      }));
    } catch (err) {
      if (err.name === "TypeError") {
        return "Window management not supported.";
      }
      return `${err.name}: ${err.message}`;
    }
  }
  return state;
}

document.querySelector("button").addEventListener("click", async () => {
  const state = await getWindowManagementPermissionState();
  document.querySelector("pre").textContent = state;
});
```

Comment (tomayac) [reply site/en/articles/multi-screen-window-placement/index.md]: I don't think it's shipping anytime soon, even if I want it to. Removed. Thank you!

**Review by rachelandrew — APPROVED** [review]

**Pull request merged**